pandas/io/common.py

```diff
@@ -460,10 +460,11 @@ def get_handle(
         handles.append(f)
 
     elif is_path:
-        if encoding:
+        is_binary = "b" in mode
+        if encoding and not is_binary:
```

Comment:
can u add a comment in what is checking here

Reply:
done